kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala

```diff
@@ -21,118 +21,144 @@ import java.util.Locale
 
 import scala.collection.JavaConverters._
 
+import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.yarn.api.records.{FinalApplicationStatus, YarnApplicationState}
 import org.apache.hadoop.yarn.client.api.YarnClient
 
-import org.apache.kyuubi.Logging
+import org.apache.kyuubi.{Logging, Utils}
 import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf.YARNClientProxyUserStrategy
+import org.apache.kyuubi.config.KyuubiConf.YARNClientProxyUserStrategy._
 import org.apache.kyuubi.engine.ApplicationOperation._
 import org.apache.kyuubi.engine.ApplicationState.ApplicationState
 import org.apache.kyuubi.engine.YarnApplicationOperation.toApplicationState
 import org.apache.kyuubi.util.KyuubiHadoopUtils
 
 class YarnApplicationOperation extends ApplicationOperation with Logging {
 
-  @volatile private var yarnClient: YarnClient = _
+  private var yarnConf: Configuration = _
+  @volatile private var adminYarnClient: Option[YarnClient] = None
   private var submitTimeout: Long = _
 
   override def initialize(conf: KyuubiConf): Unit = {
     submitTimeout = conf.get(KyuubiConf.ENGINE_YARN_SUBMIT_TIMEOUT)
-    val yarnConf = KyuubiHadoopUtils.newYarnConfiguration(conf)
-    // YarnClient is thread-safe
-    val c = YarnClient.createYarnClient()
-    c.init(yarnConf)
-    c.start()
-    yarnClient = c
-    info(s"Successfully initialized yarn client: ${c.getServiceState}")
+    yarnConf = KyuubiHadoopUtils.newYarnConfiguration(conf)
+
+    YARNClientProxyUserStrategy.withName(conf.get(KyuubiConf.YARN_PROXY_USER_STRATEGY)) match {
+      case NONE =>
+        val c = createYarnClient(yarnConf)
+        info(s"Successfully initialized admin YARN client.")
+        adminYarnClient = Some(c)
+      case ADMIN =>
+        val adminUser = conf.get(KyuubiConf.YARN_PROXY_USER_ADMIN_USER).getOrElse(Utils.currentUser)
+        Utils.doAs(adminUser) { () =>
```

Comment:
We don't need to create a proxy user if the users are the same.

Reply:
updated